tedana/metrics/dependence.py

```diff
@@ -0,0 +1,354 @@
+        # NOTE: Why only compare distributions of *unique* F-statistics?
+        noise_FT2_Z = 0.5 * np.log(F_T2_maps[comp_noise_idx, i_comp])
+        signal_FT2_Z = 0.5 * np.log(F_T2_maps[Z_clmaps[:, i_comp] == 1, i_comp]))
+        n_noise_dupls = noise_FT2_Z.size - np.unique(noise_FT2_Z).size
+        if n_noise_dupls:
+            LGR.debug('For component {}, {} duplicate noise F-values '
+                      'detected.'.format(i_comp, n_noise_dupls))
+        n_signal_dupls = signal_FT2_Z.size - np.unique(signal_FT2_Z).size
+        if n_signal_dupls:
+            LGR.debug('For component {}, {} duplicate signal F-values '
+                      'detected.'.format(i_comp, n_signal_dupes))
+        dof = np.sum(comp_noise_idx) + countsignal - 2
+
+        t_value, signal_minus_noise_p[i_comp] = stats.ttest_ind(
+            signal_FT2_Z, noise_FT2_Z, equal_var=False)
+        signal_minus_noise_z[i_comp] = t_to_z(t_value, dof)
+
+    signal_minus_noise_z[np.isnan(signal_minus_noise_t)] = 0
+    signal_minus_noise_p[np.isnan(signal_minus_noise_p)] = 0
+    return signal_minus_noise_z, signal_minus_noise_p
+
+
+def compute_signal_minus_noise_t(Z_maps, Z_cl_maps, F_T2_maps, z_thresh=1.95):
+    RepLGR.info('A t-test was performed between the distributions of T2*-model '
+                'F-statistics associated with clusters (i.e., signal) and '
+                'non-cluster voxels (i.e., noise) to generate a t-statistic '
+                '(metric signal-noise_t) and p-value (metric signal-noise_p) '
+                'measuring relative association of the component to signal '
+                'over noise.')
+    n_components = Z_maps.shape[1]
+    signal_minus_noise_t = np.zeros(n_components)
+    signal_minus_noise_p = np.zeros(n_components)
+    for i_comp in range(n_components):
+        # index voxels significantly loading on component but not from clusters
+        comp_noise_sel = ((np.abs(Z_maps[:, i_comp]) > z_thresh) &
+                          (Z_clmaps[:, i_comp] == 0))
+        comptable.loc[i_comp, 'countnoise'] = np.array(
+            comp_noise_sel, dtype=np.int).sum()
+        # NOTE: Why only compare distributions of *unique* F-statistics?
+        noise_FT2_Z = np.log10(np.unique(F_T2_maps[comp_noise_sel, i_comp]))
+        signal_FT2_Z = np.log10(np.unique(
+            F_T2_maps[Z_clmaps[:, i_comp] == 1, i_comp]))
+        (signal_minus_noise_t[i_comp],
+         signal_minus_noise_p[i_comp]) = stats.ttest_ind(
+             signal_FT2_Z, noise_FT2_Z, equal_var=False)
+
+    signal_minus_noise_t[np.isnan(signal_minus_noise_t)] = 0
+    signal_minus_noise_p[np.isnan(signal_minus_noise_p)] = 0
+    return signal_minus_noise_t, signal_minus_noise_p
+
+
+def compute_dice(Br_clmaps, F_clmaps):
+    n_components = Br_clmaps.shape[1]
+    dice_values = np.zeros(n_components)
+    for i_comp in range(n_components):
+        dice_values[i_comp] = utils.dice(Br_clmaps[:, i_comp], F_clmaps[:, i_comp])
```

Comment:
Any way we could make `utils.dice` broadcastable? I haven't looked at the function in a while but if it's possible it would be really nice!

Reply:
Should be good now.